### PR TITLE
upgrade to python 3.12

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Upgrade from Python 3.10 to 3.12

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/355
* 
# Test instructions | Instructions pour tester la modification

kick the tires on api

# Release Instructions | Instructions pour le déploiement

None.

# Reviewer checklist | Liste de vérification du réviseur

- [ ] This PR does not break existing functionality.
- [ ] This PR does not violate GCNotify's privacy policies.
- [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
- [ ] This PR does not significantly alter performance.
- [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: tests/app/db.py [hunk 1]
```diff
@@ -474,7 +474,7 @@ def create_inbound_sms(
     if not service.inbound_number:
         create_inbound_number(
             # create random inbound number
-            notify_number or "1{:10}".format(random.randint(0, 1e9 - 1)),
+            notify_number or "1{:10}".format(random.randint(0, int(1e9 - 1))),
```
[the remark this answers is not in the document]
Comment (sastels) [reply tests/app/db.py:477]: Thanks AI! ❤️

File: tests/app/notifications/test_validators.py [hunk 2]
```diff
@@ -545,7 +545,7 @@
         api_key = create_sample_api_key(notify_db, notify_db_session, service=service, key_type="normal")
 
         check_service_over_api_rate_limit_and_update_rate(service, api_key)
-        assert app.redis_store.exceeded_rate_limit.called_with("{}-{}".format(str(service.id), api_key.key_type), 3000, 60)
+        app.redis_store.exceeded_rate_limit.assert_called_with("{}-{}".format(str(service.id), api_key.key_type), 1000, 60)
```
[the remark this answers is not in the document]
Comment (sastels) [reply tests/app/notifications/test_validators.py:548]: yep, before the test was wrong and passing anyway.

**Review by jzbahrai — APPROVED** [review]

**Review by whabanks — APPROVED** [review]
LGTM! Tested the API locally by calling it, as well as hooking it up to Admin. Works smoothly!

**Comment by sastels** [comment]
lambda docker image wasn't building but fixed as described in
https://github.com/awslabs/aws-lambda-cpp/issues/124

**Pull request merged**